crates/client/l1-messages/src/contract.rs

```diff
@@ -40,14 +38,12 @@ impl TryFrom<&LogMessageToL2Filter> for HandleL1MessageTransaction {
         // L1 message nonce.
         let nonce: u64 = Felt252Wrapper::try_from(sp_core::U256(event.nonce.0))?.try_into()?;
 
-        // Add the from address here so it's directly in the calldata.
-        let mut calldata: Vec<Felt252Wrapper> = Vec::from([Felt252Wrapper::try_from(event.from_address.as_bytes())?]);
+        let mut calldata = Vec::<Felt252Wrapper>::new();
```

Comment:
You can use the iterator and map syntax. More readable here

Reply:
It would be more readable but `map()` requires `FnMut` that is not fallible... 
We need here error handling for ->  `Felt252Wrapper::try_from(sp_core::U256(x.0))?` 
It would be great if you know better solution...

Reply:
Found interesting workaround -- https://www.justanotherdot.com/posts/error-handling-with-closures-in-iterators.html

Reply:
Yep that is the way to go. You collect your `Iterator<Result<V, E>>` into a `Result<Vec<V>, E>` it will early exit at the first error encountered